### PR TITLE
Adding Microsoft.Compute 2017-12-01 schema

### PR DESCRIPTION
**Pull request by waynekuo**
Adding 2017-12-01 schema for Compute and 2018-04-01 schema for disks and snapshots.

### CODE REVIEW

File: schemas/2014-04-01-preview/deploymentTemplate.json [hunk 1]
```diff
@@ -538,6 +538,11 @@
                   { "$ref": "https://schema.management.azure.com/schemas/2017-11-01/Microsoft.Network.json#/resourceDefinitions/networkSecurityGroups" },
                   { "$ref": "https://schema.management.azure.com/schemas/2017-11-01/Microsoft.Network.json#/resourceDefinitions/networkInterfaces" },
                   { "$ref": "https://schema.management.azure.com/schemas/2017-11-01/Microsoft.Network.json#/resourceDefinitions/routeTables" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2017-03-30/Microsoft.Compute.json#/resourceDefinitions/images" },
```
Comment (TianoMS) [inline schemas/2014-04-01-preview/deploymentTemplate.json]: >2017-03-30 [](start = 73, length = 10)

is this correct? should it be "2017-12-01"?

File: schemas/2018-04-01/Microsoft.Compute.json [hunk 2]
```diff
@@ -0,0 +1,432 @@
+{
+  "id": "https://schema.management.azure.com/schemas/2018-04-01/Microsoft.Compute.json#",
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "Microsoft.Compute",
+  "description": "Microsoft Compute Resource Types",
+  "resourceDefinitions": {
+    "disks": {
```
Comment (TianoMS) [inline schemas/2018-04-01/Microsoft.Compute.json]: >disks [](start = 5, length = 5)

also add some tests for 2018-04-01 schemas?

**Comment by Tiano2017** [comment]
@waynekuo there are some conflicts. I also left a few comments. Please take a look.

File: schemas/2018-04-01/Microsoft.Compute.json (hunk 2 above)
Comment (waynekuo) [reply schemas/2018-04-01/Microsoft.Compute.json]: Yes, still to come

File: schemas/2014-04-01-preview/deploymentTemplate.json (hunk 1 above)
Comment (waynekuo) [reply schemas/2014-04-01-preview/deploymentTemplate.json]: Fixed

File: schemas/2014-04-01-preview/deploymentTemplate.json [hunk 3]
```diff
@@ -563,7 +568,8 @@
                   { "$ref": "https://schema.management.azure.com/schemas/2016-12-01/Microsoft.RecoveryServices.json#/resourceDefinitions/vaults_backupFabrics_protectionContainers_protectedItems" },
                   { "$ref": "https://schema.management.azure.com/schemas/2016-12-01/Microsoft.RecoveryServices.json#/resourceDefinitions/vaults_backupPolicies" },
                   { "$ref": "https://schema.management.azure.com/schemas/2017-07-01/Microsoft.RecoveryServices.json#/resourceDefinitions/vaults_backupFabrics_backupProtectionIntent" },
-                  { "$ref": "#/definitions/resourceGeneric" }
+                  { "$ref": "https://schema.management.azure.com/schemas/2018-04-01/Microsoft.Compute.json#/resourceDefinitions/disks" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2018-04-01/Microsoft.Compute.json#/resourceDefinitions/snapshots" },                  { "$ref": "#/definitions/resourceGeneric" }
```
Comment (TianoMS) [inline schemas/2014-04-01-preview/deploymentTemplate.json]: >s" },                  { "$r [](start = 136, length = 28)

please fix the format.

File: schemas/2015-01-01/deploymentTemplate.json [hunk 4]
```diff
@@ -567,8 +572,9 @@
...
                   { "$ref": "https://schema.management.azure.com/schemas/2016-12-01/Microsoft.RecoveryServices.json#/resourceDefinitions/vaults_backupFabrics_protectionContainers_protectedItems" },
                   { "$ref": "https://schema.management.azure.com/schemas/2016-12-01/Microsoft.RecoveryServices.json#/resourceDefinitions/vaults_backupPolicies" },
-                  { "$ref": "https://schema.management.azure.com/schemas/2017-07-01/Microsoft.RecoveryServices.json#/resourceDefinitions/vaults_backupFabrics_backupProtectionIntent" }
-                ]
+                  { "$ref": "https://schema.management.azure.com/schemas/2017-07-01/Microsoft.RecoveryServices.json#/resourceDefinitions/vaults_backupFabrics_backupProtectionIntent" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2018-04-01/Microsoft.Compute.json#/resourceDefinitions/disks" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2018-04-01/Microsoft.Compute.json#/resourceDefinitions/snapshots" }                ]
```
Comment (TianoMS) [inline schemas/2015-01-01/deploymentTemplate.json]: >           ] [](start = 145, length = 12)

format issues here.

File: tests/2017-12-01/Microsoft.Compute.tests.json [hunk 5]
```diff
@@ -0,0 +1,1329 @@
...
+        "properties": {
+          "platformFaultDomainCount": 2,
+          "platformUpdateDomainCount": 2,
+          "managed": true
+        }
+      }
+    },
+    {
+      "name": "Managed Data Disk",
+      "definition": "https://schema.management.azure.com/schemas/2018-04-01/Microsoft.Compute.json#/resourceDefinitions/disks",
```
Comment (TianoMS) [inline tests/2017-12-01/Microsoft.Compute.tests.json]: >2018-04-01 [](start = 65, length = 10)

can we move the tests for 2018-04-01 to a separate folder?

Comment (waynekuo) [reply tests/2017-12-01/Microsoft.Compute.tests.json]: Sure

File: schemas/2014-04-01-preview/deploymentTemplate.json (hunk 3 above)
Comment (waynekuo) [reply schemas/2014-04-01-preview/deploymentTemplate.json]: Fixed

File: schemas/2015-01-01/deploymentTemplate.json (hunk 4 above)
Comment (waynekuo) [reply schemas/2015-01-01/deploymentTemplate.json]: Fixed

**Pull request merged**